opm/autodiff/NonlinearSolver_impl.hpp

```diff
@@ -260,24 +260,28 @@ namespace Opm
         dxOld = dx;
 
         switch (relaxType()) {
-            case DAMPEN:
+            case DAMPEN: {
                 if (omega == 1.) {
                     return;
                 }
-                for (size_t i = 0; i < dx.size(); ++i) {
+                auto i = dx.size();
+                for (i = 0; i < dx.size(); ++i) {
                     dx[i] *= omega;
                 }
                 return;
-            case SOR:
+            }
+            case SOR: {
                 if (omega == 1.) {
                     return;
                 }
-                for (size_t i = 0; i < dx.size(); ++i) {
+                auto i = dx.size();
+                for (i = 0; i < dx.size(); ++i) {
```

Comment:
Exotic?  It's been around for more than five years now.  I think that in this setting, where it's quite obvious what the return type is, the `auto` is absolutely perfect.